### PR TITLE
Make height and width optional

### PR DESCRIPTION
**Pull request by patrickvonplaten**

### CODE REVIEW

File: src/diffusers/pipelines/versatile_diffusion/pipeline_versatile_diffusion_dual_guided.py [hunk 1]
```diff
@@ -551,6 +551,9 @@ def __call__(
             [`~pipelines.stable_diffusion.ImagePipelineOutput`] if `return_dict` is True, otherwise a `tuple. When
             returning a tuple, the first element is a list with the generated images.
         """
+        # 0. Default height and width to unet
+        height = height or self.unet.config.sample_size * 8
+        width = width or self.unet.config.sample_size * 8
```
Comment (anton-l) [inline src/diffusers/pipelines/versatile_diffusion/pipeline_versatile_diffusion_dual_guided.py:556]:
```suggestion
        height = height or self.image_unet.config.sample_size * 8
        width = width or self.image_unet.config.sample_size * 8
```

File: src/diffusers/pipelines/versatile_diffusion/pipeline_versatile_diffusion_image_variation.py [hunk 2]
```diff
@@ -391,6 +391,9 @@ def __call__(
             list of `bool`s denoting whether the corresponding generated image likely represents "not-safe-for-work"
             (nsfw) content, according to the `safety_checker`.
         """
+        # 0. Default height and width to unet
+        height = height or self.unet.config.sample_size * 8
+        width = width or self.unet.config.sample_size * 8
```
Comment (anton-l) [inline src/diffusers/pipelines/versatile_diffusion/pipeline_versatile_diffusion_image_variation.py:396]:
```suggestion
        # 0. Default height and width to unet
        height = height or self.image_unet.config.sample_size * 8
        width = width or self.image_unet.config.sample_size * 8
```

File: src/diffusers/pipelines/versatile_diffusion/pipeline_versatile_diffusion_text_to_image.py [hunk 3]
```diff
@@ -443,6 +443,9 @@ def __call__(
             list of `bool`s denoting whether the corresponding generated image likely represents "not-safe-for-work"
             (nsfw) content, according to the `safety_checker`.
         """
+        # 0. Default height and width to unet
+        height = height or self.unet.config.sample_size * 8
+        width = width or self.unet.config.sample_size * 8
```
Comment (anton-l) [inline src/diffusers/pipelines/versatile_diffusion/pipeline_versatile_diffusion_text_to_image.py:448]:
```suggestion
        # 0. Default height and width to unet
        height = height or self.image_unet.config.sample_size * 8
        width = width or self.image_unet.config.sample_size * 8
```

**Review by pcuenca — APPROVED** [review]
Looks good!

**Review by anton-l — APPROVED** [review]
The fast tests catch all of the previously mismatched sample_size configs now, but otherwise looks great!

**Pull request merged**